### DISCUSSION TITLE
Bug fixes for legacy versions

### DISCUSSION THREAD

**Discussion by TheLimeGlass** [SkriptLang/Skript#5218] [General]
# Hello Skripters 👋
Today we're releasing version 2.6.4 which brings out a couple fixes towards Minecraft versions below 1.14

# 📢 End of 1.12.2 Support
As mentioned in the previous releases, we have decided to drop support for 1.12 and lower (legacy versions) in **Skript 2.7**.
This means that **Skript 2.6.X** is the last version that will work with the legacy versions.
_Only critical fixes will be backported to 2.6.X._

# ⭐️ 2.6.4 Changelog
Click below to view the entire changelog for this release.
<details>
<summary>
<i>Full changelog</i>
</summary>

## Bug Fixes
- Fixed a bug where if the lifetime value of a firework launch was greater than 127 or less than 0, it would throw an error. (thanks @DelayedGaming)
- Fixed a bug where the offhand tool of player was incorrect in the PlayerHeldItem event. (https://github.com/SkriptLang/Skript/issues/5094 thanks @TUCAOEVER)
- Fixed variable names with long numbers either taking long parsing time, or not being able to compare with another similar number. (https://github.com/SkriptLang/Skript/issues/4729, https://github.com/SkriptLang/Skript/pull/3929)
- Fixed ExprTimePlayed not working on versions 1.13-1.14 (https://github.com/SkriptLang/Skript/issues/4929)
- Fixed VariableString#isQuoted to work correctly, should be returning false for the input string " (with withQuotes=true). (#5149)
- Fixed a bug where the component type of the returned array from ExprEntities not being the return type (e.g. Player), but a generic Entity instead, in the case that the syntax was used with radius, and no entities matched the query. (#5124)
- Fixed a bunch of issues relating to conversion for list-modifying expressions.
- Fixed an issue with how EffPotion interacted with patterns for `potioneffects` and `potioneffecttypes` (#5117) (thanks @Fusezion)
- Fixed EffBroadcast not working on 1.9.4 (#5019)
- Fixed a bug where you couldn't use colons in the ExprParse syntax (thanks @Mr-Darth)
- Fixed a NPE from happening when spawning a falling block without specifying the block type and will default to Stone (https://github.com/SkriptLang/Skript/issues/5042) (thanks @UnderscoreTud)
- Fixed `[on] silverfish enter` and `[on] silverfish exit` events not actually working.
- Changed item serialization with blocks to properly serialize. **This means you cannot downgrade your Skript version after this version without losing those variables storing block value data!**
- Changed some event-values to return as an ItemStack rather than Skript's own ItemType to avoid some issues. (https://github.com/SkriptLang/Skript/issues/4109)
- Fixed a bug where Skript couldn't spawn lingering potions in 1.9-1.13 (https://github.com/SkriptLang/Skript/issues/4933)
- Fixed a bug where Skript might not start on 1.11.2 due to enchantment sweeping edge not being registered (https://github.com/SkriptLang/Skript/issues/4933)
- Fixed a bug where Skript wouldn't register sound syntaxes on 1.10.2 due to sound category not existing (https://github.com/SkriptLang/Skript/issues/4933)
- Fixed a NPE when using `of %itemtype%` syntax in preparing craft event (https://github.com/SkriptLang/Skript/issues/4959)
- Fixed an issue where EffConnect would error due to Bukkit#getOnlinePlayers not returning anything, yet a player instance was provided. ([#5189](https://github.com/SkriptLang/Skript/issues/5187))

[Click here to view the full list of commits made since 2.6.3](https://github.com/SkriptLang/Skript/compare/2.6.3...2.6.4)

</details>

## Help Us Test
We have an [official Discord community](https://discord.gg/ZPsZAg6ygu) for beta testing Skript's new features and releases.

# ⭐️ Thank you
We saw a large increase in new contributors recently, and we just wanted to thank all those who have contributed to this version of Skript, lots of issues and suggestions arise and we couldn't have done it all without the help of the community.

As always, if you encounter any issues or suggestions please report them at https://github.com/SkriptLang/Skript/issues
If you have any ideas or input for the future development of Skript, you can share those too at https://github.com/SkriptLang/Skript/discussions

<hr /><em>This discussion was created from the release <a href='https://github.com/SkriptLang/Skript/releases/tag/2.6.4'>Bug fixes for legacy versions</a>.</em>

**Comment by loolkkekrr** [comment]
Write a comment

**Reply by loolkkekrr** [reply]
Write a reply

**Comment by SpellDigger** [comment]
Boi when Its gonna be in spigotmc That is not download via external site 💀

**Comment by ByteAfterlife** [comment]
I love skript
